app/jobs/import_versions_job.rb

```diff
@@ -24,20 +29,25 @@ def import_raw_data(raw_data)
         import_record(record)
       rescue Api::NotImplementedError => error
         @import.processing_errors << "Row #{row}: #{error.message}"
+        Rails.logger.error "Row #{row}: #{error.message}"
       rescue Api::InputError => error
         @import.processing_errors << "Row #{row}: #{error.message}"
+        Rails.logger.error "Row #{row}: #{error.message}"
       rescue ActiveModel::ValidationError => error
         messages = error.model.errors.full_messages.join(', ')
         @import.processing_errors << "Row #{row}: #{messages}"
+        Rails.logger.error "Row #{row}: #{messages}"
       rescue ActiveRecord::RecordInvalid => error
         messages = error.model.errors.full_messages.join(', ')
         @import.processing_errors << "Row #{row}: #{messages}"
+        Rails.logger.error "Row #{row}: #{messages}"
```

Comment:
So with these errors, I very explicitly chose *not* to log (totally agree I should have for the unknown type of error), because they represent user input problems that a maintainer of the system should be bothered with or worry about. They’re only really useful as feedback to the user who imported data because they concern errors that user (*not* the app authors/maintainers) need to address.

Reply:
(If following via e-mail, I missed a critical “not” in “a maintainer of the system should **not** be bothered.” Edited the comment on GH.)